@here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts

```diff
@@ -194,34 +222,41 @@ export class VersionedLayerClient {
 
         const quadKey = dataRequest.getQuadKey();
         if (quadKey) {
-            const quadKeyPartitionsRequest = new QuadKeyPartitionsRequest()
-                .withQuadKey(quadKey)
-                .withVersion(usingVersion);
+            const mortonCode = `${mortonCodeFromQuadKey(quadKey)}`;
+            const noContentError = new HttpError(
+                204,
+                `No dataHandle for quadKey {column: ${quadKey.column}, row: ${quadKey.row}, level: ${quadKey.level}}. HRN: ${this.hrn}`
+            );
 
-            const quadTreeIndexResponse = await this.getPartitions(
-                quadKeyPartitionsRequest
-            ).catch(error => Promise.reject(error));
+            const quadTreeIndexResponse = await fetchQuadTreeIndex({
+                catalogHrn: HRN.fromString(this.hrn),
+                catalogVersion: usingVersion,
+                fetchOptions: dataRequest.getFetchOption(),
+                layerId: this.layerId,
+                layerType: "versioned",
+                quadKey,
+                settings: this.settings,
+                abortSignal,
+                billingTag: dataRequest.getBillingTag()
+            });
 
-            if (
-                quadTreeIndexResponse.status &&
-                quadTreeIndexResponse.status === 400
-            ) {
-                return Promise.reject(quadTreeIndexResponse);
+            if (!quadTreeIndexResponse.subQuads) {
+                return Promise.reject(noContentError);
             }
 
-            return quadTreeIndexResponse.subQuads &&
-                quadTreeIndexResponse.subQuads.length
-                ? this.downloadPartition(
-                      quadTreeIndexResponse.subQuads[0].dataHandle,
-                      abortSignal,
-                      dataRequest.getBillingTag()
-                  )
-                : Promise.reject(
-                      new HttpError(
-                          204,
-                          `No dataHandle for quadKey {column: ${quadKey.column}, row: ${quadKey.row}, level: ${quadKey.level}}. HRN: ${this.hrn}`
-                      )
-                  );
+            const subQuad = quadTreeIndexResponse.subQuads.find(
+                item => item.subQuadKey === mortonCode
+            );
+
+            if (!subQuad) {
+                return Promise.reject(noContentError);
+            }
```

Comment:
What about checking the parent next? Note that subQuads array could be empty

Reply:
it's per Andrey's request. Checking parents is in getAggregatedData only

Reply:
"Note that subQuads array could be empty" in this case subQuad will be undefined